pkg/reconciler/pipeline/controller.go

```diff
@@ -985,6 +1000,34 @@ func (r *pipelineReconciler) updateVerticeDesiredPhase(ctx context.Context, pl *
 	return isVertexPatched, nil
 }
 
+func (r *pipelineReconciler) removeVertexReplicas(ctx context.Context, vertex *dfv1.Vertex) error {
```

Comment:
Can make it one call to patch desiredPhase and replicas to the vertex objects? That will reduce the chance of race conditions.